internal/services/healthcare/healthcare_dicom_resource.go

```diff
@@ -153,10 +246,38 @@ func resourceHealthcareApisDicomServiceCreate(d *pluginsdk.ResourceData, meta in
 
 	t := d.Get("tags").(map[string]interface{})
 
+	var enableDataPartitions *bool
+	if v, ok := d.GetOk("data_partitions_enabled"); ok {
+		enableDataPartitions = pointer.To(v.(bool))
+	}
+
+	var corsConfiguration *dicomservices.CorsConfiguration
+	if v, ok := d.GetOk("cors"); ok {
+		corsConfiguration = expandDicomServiceCorsConfiguration(v.([]interface{}))
+	}
+
+	var encryption *dicomservices.Encryption
+	if v, ok := d.GetOk("encryption_key_url"); ok {
+		encryption = &dicomservices.Encryption{
+			CustomerManagedKeyEncryption: &dicomservices.EncryptionCustomerManagedKeyEncryption{
+				KeyEncryptionKeyURL: pointer.To(v.(string)),
+			},
+		}
+	}
+
+	var storageConfiguration *dicomservices.StorageConfiguration
+	if v, ok := d.GetOk("storage"); ok {
+		storageConfiguration = expandStorageConfiguration(v.([]interface{}))
+	}
+
 	parameters := dicomservices.DicomService{
 		Identity: i,
 		Properties: &dicomservices.DicomServiceProperties{
-			PublicNetworkAccess: pointer.To(dicomservices.PublicNetworkAccessEnabled),
+			CorsConfiguration:    corsConfiguration,
+			EnableDataPartitions: enableDataPartitions,
+			Encryption:           encryption,
+			PublicNetworkAccess:  pointer.To(dicomservices.PublicNetworkAccessEnabled),
+			StorageConfiguration: storageConfiguration,
 		},
 		Location: pointer.To(location.Normalize(d.Get("location").(string))),
 		Tags:     tags.Expand(t),
```

Comment:
We typically don't set optional properties in the create payload if they user hasn't specified them, so this would look more like:
```
	parameters := dicomservices.DicomService{
		Identity: i,
		Properties: &dicomservices.DicomServiceProperties{
			PublicNetworkAccess:  pointer.To(dicomservices.PublicNetworkAccessEnabled),
		},
		Location: pointer.To(location.Normalize(d.Get("location").(string))),
		Tags:     tags.Expand(t),
	}

	if v, ok := d.GetOk("data_partitions_enabled"); ok {
		parameters.Properties.EnableDataPartitions = pointer.To(v.(bool))
	}

	cors := expandDicomServiceCorsConfiguration(d.Get("cors").([]interface{}))
	if cors != nil {
		parameters.Properties.CorsConfiguration = cors
	}

	if v, ok := d.GetOk("encryption_key_url"); ok && v != "" {
		parameters.Properties.Encryption = &dicomservices.Encryption{
			CustomerManagedKeyEncryption: &dicomservices.EncryptionCustomerManagedKeyEncryption{
				KeyEncryptionKeyURL: pointer.To(v.(string)),
			},
		}
	}

	storage := expandStorageConfiguration(d.Get("storage").([]interface{}))
	if storage != nil  {
		parameters.Properties.StorageConfiguration = storage
	}

```